runelite-client/src/main/java/net/runelite/client/plugins/npcunaggroarea/NpcAggroAreaPlugin.java

```diff
@@ -0,0 +1,554 @@
+		int npcLvl = composition.getCombatLevel();
+		String npcName = composition.getName().toLowerCase();
+		if (npcLvl > 0 && playerLvl > npcLvl * 2 && !isInWilderness(npc.getWorldLocation()))
+		{
+			return false;
+		}
+
+		for (String pattern : npcNamePatterns)
+		{
+			if (WildcardMatcher.matches(pattern, npcName))
+			{
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	private void checkAreaNpcs(final NPC... npcs)
+	{
+		if (active)
+		{
+			return;
+		}
+
+		for (NPC npc : npcs)
+		{
+			if (npc == null)
+			{
+				continue;
+			}
+
+			if (isNpcMatch(npc))
+			{
+				active = true;
+				break;
+			}
+		}
+
+		reevaluateActive();
+	}
+
+	private void recheckActive()
+	{
+		active = config.alwaysActive();
+		checkAreaNpcs(client.getCachedNPCs());
+	}
+
+	@Subscribe
+	public void onNpcSpawned(NpcSpawned event)
+	{
+		if (config.alwaysActive())
+		{
+			return;
+		}
+
+		checkAreaNpcs(event.getNpc());
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		WorldPoint newLocation = client.getLocalPlayer().getWorldLocation();
+		if (lastPlayerLocation != null)
+		{
+			if (safeCenters[1] == null && newLocation.distanceTo2D(lastPlayerLocation) > SAFE_AREA_RADIUS * 4)
+			{
+				safeCenters[0] = null;
+				safeCenters[1] = newLocation;
+				resetTimer();
+				calculateLinesToDisplay();
+
+				// We don't know where the previous area was, so if the player e.g.
+				// entered a dungeon and then goes back out, he/she may enter the previous
+				// area which is unknown and would make the plugin inaccurate
+				previousUnknownCenter = lastPlayerLocation;
+			}
+		}
+
+		if (safeCenters[0] == null && previousUnknownCenter != null &&
+			previousUnknownCenter.distanceTo2D(newLocation) <= UNKNOWN_AREA_RADIUS)
+		{
+			// Player went back to their previous unknown area before the 2nd
+			// center point was found, which means we don't know where it is again.
+			safeCenters[1] = null;
+			removeTimer();
+			calculateLinesToDisplay();
+		}
+
+		if (safeCenters[1] != null)
+		{
+			if (Arrays.stream(safeCenters).noneMatch(
+				x -> x != null && x.distanceTo2D(newLocation) <= SAFE_AREA_RADIUS))
+			{
+				safeCenters[0] = safeCenters[1];
+				safeCenters[1] = newLocation;
+				resetTimer();
+				calculateLinesToDisplay();
+				previousUnknownCenter = null;
+			}
+		}
+
+		lastPlayerLocation = newLocation;
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		String key = event.getKey();
+		switch (key)
+		{
+			case "npcUnaggroAlwaysActive":
+				recheckActive();
+				break;
+			case "npcUnaggroShowTimer":
+				if (currentTimer != null)
+				{
+					currentTimer.setVisible(active && config.showTimer());
+				}
+				break;
+			case "npcUnaggroCollisionDetection":
+			case "npcUnaggroShowAreaLines":
+				calculateLinesToDisplay();
+				break;
+			case "npcUnaggroNames":
+				npcNamePatterns = NAME_SPLITTER.splitToList(config.npcNamePatterns());
+				recheckActive();
+				break;
+		}
+	}
+
+	private void loadConfig()
```

Comment:
Maybe after @Adam- requested removal of per-account settings this should not be persisted at all.

Reply:
Why? Removing it would cause the plugin to not work on login until the player teleports.

Reply:
Because atm it would break also when using mutliple accounts on same PC.

Reply:
So rather than breaking on a subset of users it should break for everyone?

Reply:
Or the change that @Adam- requested should be put back.